source/connection/connection-pools.txt

```diff
@@ -0,0 +1,160 @@
+.. _java-connection-pools:
+
+================
+Connection Pools
+================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+Overview
+--------
+
+In this guide, you can learn about how {+driver-short+} uses connection pools to manage
+connections to a MongoDB deployment and how you can configure connection pool settings
+in your application.
+
+A connection pool is a cache of open database connections maintained by {+driver-short+}.
+When your application requests a connection to MongoDB, {+driver-short+} seamlessly
+gets a connection from the pool, performs operations, and returns the connection
+to the pool for reuse.
+
+Connection pools help reduce application latency and the number of times new connections
+are created by {+driver-short+}.
+
+Create a Connection Pool
+------------------------
+
+Every ``MongoClient`` instance has a built-in connection pool for each server 
+in your MongoDB topology. Connection pools open sockets on demand to support 
+concurrent MongoDB operations in your multi-threaded application.
+
+The ``maxPoolSize`` option sets the maximum size of each connection pool, which 
+defaults to 100. If the number of in-use connections to a server reaches the 
+value of maxPoolSize, the next request to that server will wait until a 
+connection becomes available.
+
+Each ``MongoClient`` instance opens two more sockets per server in your MongoDB 
+topology for monitoring the server's state.
+
+Configure a Connection Pool
+---------------------------
+
+You can specify settings for your connection pool using either a connection
+string or by passing a ``MongoClientSettings`` object to the 
+``MongoClients.create()`` method.
+
+The following code creates a client with a maximum connection pool size of ``50``. 
+Select the :guilabel:`Connection String` or :guilabel:`MongoClientSettings`  tab to
+see the corresponding syntax:
+
+.. tabs::
+
+  .. tab:: Connection String
+    :tabid: uri
+    
+    .. code-block:: java
+
+      ConnectionString connectionString = "mongodb://<host>:<port>/?maxPoolSize=50"
+      MongoClient mongoClient = MongoClients.create(connectionString)
+
+    The following are connection string settings you can use to configure your 
+    connection pool:
+
+    .. list-table::
+      :widths: 25,75                                                          
+      :header-rows: 1
+
+      * - Setting
+        - Description
+        
+      * - ``connectTimeoutMS``
+
+        - Specifies the maximum amount of time, in milliseconds, the Java driver 
+          waits for a connection to open before timing out. A value of 0 instructs 
+          the driver to never time out while waiting for a connection to open.
```

Comment:
This is true, but in practice the OS has a connect timeout so it won't really wait forever.  See if you can find a reference for that somewhere?

Reply:
Might be too general to find a resource to link to. Edited to "A value of 0 instructs the driver to default to your operating system's connection timeout value". What do you think? I can also remove the sentence entirely.

Reply:
Removed from the table instead, due to previous comment.